tests/expiringdict_test.py

```diff
@@ -1,127 +1,130 @@
 from time import sleep
 
 from mock import Mock, patch
-from nose.tools import assert_raises, eq_, ok_
 
 from expiringdict import ExpiringDict
 
+from pytest import raises
+
 
 def test_create():
-    assert_raises(AssertionError, ExpiringDict, max_len=1, max_age_seconds=-1)
-    assert_raises(AssertionError, ExpiringDict, max_len=0, max_age_seconds=1)
+    with raises(AssertionError):
+        ExpiringDict(max_len=1, max_age_seconds=-1)
+        ExpiringDict(max_len=0, max_age_seconds=1)
 
     d = ExpiringDict(max_len=3, max_age_seconds=0.01)
-    eq_(len(d), 0)
+    assert len(d) == 0
 
 
 def test_basics():
     d = ExpiringDict(max_len=3, max_age_seconds=0.01)
 
-    eq_(d.get('a'), None)
+    assert d.get('a') is None
     d['a'] = 'x'
-    eq_(d.get('a'), 'x')
+    assert d.get('a') == 'x'
 
     sleep(0.01)
-    eq_(d.get('a'), None)
+    assert d.get('a') is None
 
     d['a'] = 'y'
-    eq_(d.get('a'), 'y')
+    assert d.get('a') == 'y'
 
-    ok_('b' not in d)
+    assert 'b' not in d
     d['b'] = 'y'
-    ok_('b' in d)
+    assert 'b' in d
 
     sleep(0.01)
-    ok_('b' not in d)
+    assert 'b' not in d
 
     # a is still in expiringdict, next values should expire it
     d['c'] = 'x'
     d['d'] = 'y'
     d['e'] = 'z'
 
     # dict if full
-    ok_('c' in d)
-    ok_('d' in d)
+    assert 'c' in d
+    assert 'd' in d
 
     d['f'] = '1'
     # c should gone after that
-    ok_('c' not in d, 'Len of dict is more than max_len')
+    assert 'c' not in d, 'Len of dict is more than max_len'
 
     # test __delitem__
     del d['e']
-    ok_('e' not in d)
+    assert 'e' not in d
 
 
 def test_pop():
     d = ExpiringDict(max_len=3, max_age_seconds=0.01)
     d['a'] = 'x'
-    eq_('x', d.pop('a'))
+    assert 'x' == d.pop('a')
     sleep(0.01)
-    eq_(None, d.pop('a'))
+    assert d.pop('a') is None
 
 
 def test_repr():
     d = ExpiringDict(max_len=2, max_age_seconds=0.01)
     d['a'] = 'x'
-    eq_(str(d), "ExpiringDict([('a', 'x')])")
+    assert str(d) == "ExpiringDict([('a', 'x')])"
```

Comment:
The format changes starting with Python 3.12:
```suggestion
    assert str(d) == "ExpiringDict({'a': 'x')}"
```

Reply:
The empty representation has changed too:

```diff
-    assert str(d) == "ExpiringDict([])"
+    assert str(d) == "ExpiringDict()"
```

But stepping back, the module itself [isn’t compatible with Python 3.12](https://github.com/mailgun/expiringdict/issues/55) so I don’t think it’s appropriate to update the tests for that yet.